File: src/main/java/ru/chernov/urlshortener/converter/UuidConverter.java
package ru.chernov.urlshortener.converter;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import java.util.UUID;


@Converter(autoApply = true)
public class UuidConverter implements AttributeConverter<UUID, String> {

    @Override
    public String convertToDatabaseColumn(UUID uuid) {
        return uuid.toString();
    }


    @Override
    public UUID convertToEntityAttribute(String value) {
        return UUID.fromString(value);
    }

}

File: src/main/kotlin/ru/chernov/urlshortener/converter/auto/UUIDConverter.kt
package ru.chernov.urlshortener.converter.auto

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.util.*


@Converter(autoApply = true)
class UUIDConverter : AttributeConverter<UUID?, String?> {

    override fun convertToDatabaseColumn(uuid: UUID?): String? {
        return uuid?.toString()
    }


    override fun convertToEntityAttribute(string: String?): UUID? {
        return string?.let { UUID.fromString(string) }
    }

}